comictaggerlib/main.py

```diff
@@ -148,20 +164,18 @@ def main(self) -> None:
                 print("Key set")  # noqa: T201
                 return
 
-        try:
-            talker_api = ct_api.get_comic_talker("comicvine")(  # type: ignore[call-arg]
-                version=version,
-                cache_folder=self.options[0].runtime_config.user_cache_dir,
-                series_match_thresh=self.options[0].comicvine_series_match_search_thresh,
-                remove_html_tables=self.options[0].comicvine_remove_html_tables,
-                use_series_start_as_volume=self.options[0].comicvine_use_series_start_as_volume,
-                wait_on_ratelimit=self.options[0].autotag_wait_and_retry_on_rate_limit,
-                api_url=self.options[0].comicvine_cv_url,
-                api_key=self.options[0].comicvine_cv_api_key,
-            )
-        except TalkerError as e:
-            logger.exception("Unable to load talker")
-            error = (str(e), True)
+        talker_api = load_talker("comicvine")
+        if not talker_api:
+            if self.options[0].runtime_no_gui:
+                # Need to hard exit on talker error with the CLI
+                error = (
+                    "Unable to load talker. Please check your source name ID is correct in your config file or your command line option.",
+                    True,
```

Comment:
Move all of this and `def load_talker(talker):` to it's own method. Directly return the error

Reply:
As https://github.com/comictagger/comictagger/pull/419 (plugin settings) replaces this anyway, I'll leave this and make any additional changes required after the plugin settings is merged.